src/CoreFoundation/CFArray.cs

```diff
@@ -128,6 +128,20 @@ internal static nint GetCount (IntPtr array)
 		extern static CFArrayRef CFArrayCreateCopy (CFAllocatorRef allocator, CFArrayRef theArray);
 
 		internal CFArray Clone () => new CFArray (CFArrayCreateCopy (IntPtr.Zero, GetCheckedHandle ()), true);
+
+		// identical signature to NSArray API
+		static public string?[]? StringArrayFromHandle (IntPtr handle)
+		{
+			if (handle == IntPtr.Zero)
+				return null;
+
+			var c = CFArrayGetCount (handle);
+			string?[] ret = new string [c];
+
+			for (nint i = 0; i < c; i++)
+				ret [i] = CFString.FromHandle (CFArrayGetValueAtIndex (handle, i));
```

Comment:
I wonder if it would be faster to call `CFArrayGetValues` to get all the values at the same time (or chunks of them at least)?

Reply:
Good catch! Most probably, IOW when the element count is large enough to offset the temporary buffer allocation.

I'll try it (on another night) for `ArrayFromHandle` and update `StringArrayFromHandle` if that's the case.